include/cgf_diagnostics.h

```diff
@@ -90,11 +91,11 @@ class cgf_diagnostics {
 		m_is_checking = false;
 		m_expected_target = std::nullopt;
 
-		for(size_t i = 0; i < m_expected_buffer_accesses->get_num_accesses(); ++i) {
+		for(size_t i = 0; i < m_expected_buffer_accesses->size(); ++i) {
 			if(!m_registered_buffer_accesses[i]) {
 				throw std::runtime_error(fmt::format("Accessor {} for buffer {} is not being copied into the kernel. This indicates a bug. Make sure "
 				                                     "the accessor is captured by value and not by reference, or remove it entirely.",
-				    i, m_expected_buffer_accesses->get_nth_access(i).first));
+				    i, m_expected_buffer_accesses->operator[](i).bid));
```

Comment:
You could just use `buffer_accesses` here to avoid the odd `->operator[]()` call.

Reply:
Right!